code/client/src/pages/Show/TransactionViewer.jsx

```diff
@@ -0,0 +1,241 @@
+  NativeCrossShardTransfer: 'NativeCrossShardTransfer', // his operation is for transferring native ONE tokens between different shards. Note that cross-shard transfer operations appear in pairs, but not in the same transaction/block. The first appearance will be on the submitted transaction on the source/from shard. The second appearance will be on the destination/to shard.
+  ContractCreation: 'ContractCreation', // This operation creates/instantiates a smart contract. All "ContractCreation" operations come in pairs, one operation deducting the native ONE tokens from the sender's account, the other operation crediting said amount to the contract's account (as dictated by the contract instantiation code).
+  Genesis: 'Genesis', // This is a special operation that is only valid for block 0. It indicates the initial funds of the shard.
+  UndelegationPayout: 'UndelegationPayout', // This is a special operation that is only present on the last block of an Epoch. It represents any delegated/locked native ONE tokens that are refunded to an account.
+  CreateValidator: 'CreateValidator', // This operation creates a validator on the Harmony network for the sender's account.
+  EditValidator: 'EditValidator', // This operation edits the sender's validator information or election status (i.e: validator name and/or being eligible for election).
+  Delegate: 'Delegate', // This operation delegates (or re-delegates from pending undelegations) native ONE tokens from the sender's account to a validator's account.
+  Undelegate: 'Undelegate', // This operation removes delegation from a validator. Note that this operation does not immediately unlock the sender's tokens.
+  CollectRewards: 'CollectRewards', // This operation credits all the staking rewards to the sender's account.
+}
+
+const PAGE_SIZE = 10
+
+function formatOneValue (val) {
+  const bi = BigInt(val) / BigInt(10 ** 14)
+  return parseInt(bi.toString()) / 10000
+}
+
+function getOperationInfo (tx, address) {
+  if (tx.value > 0 || !tx.operations) {
+    return { value: formatOneValue(tx.value) + ' ONE', type: tx.to === address ? 'Receive' : tx.from === address ? 'Send' : 'Unknown' }
+  }
+  let meaningfulOp
+  for (const op of tx.operations) {
+    if (op.type === OperationType.Gas) continue
+    if (op.type === OperationType.NativeTransfer && op.account.address === address) {
+      meaningfulOp = op
+    }
+  }
+  let opType
+  let amount
+  if (tx.operations?.length) {
+    opType = tx.operations.find(op => op.type && op.type !== OperationType.Gas)?.type
+  }
+  if (!meaningfulOp) {
+    opType = tx.to === address ? 'Receive' : tx.from === address ? 'Send' : 'Unknown'
```

Comment:
1wallet addresses are always `to`, and `from` is always the relayer's address. So this way of parsing is incorrect